client/test/e2e/enterprises/enterprises.spec.js

```diff
@@ -0,0 +1,83 @@
+/*global describe, it, element, by, beforeEach, inject, browser */
+
+var chai = require('chai');
+var chaiAsPromised = require('chai-as-promised');
+chai.use(chaiAsPromised);
+var expect = chai.expect;
+
+var FormUtils = require('../shared/FormUtils');
+
+describe('The Enterprises Module', function () {
+
+  // shared methods
+  var path = '#/enterprises';
+  var ENTERPRISES = {
+    name : 'Interchurch Medical Assistance',
+    abbr : 'IMA', 
+    email : 'ima@imaworldhealth.com', 
+    po_box : 'POBOX USA 1', 
+    phone : '01500', 
+    currency_id : 1
+  };
+
+  function update(n) {
+    return element(by.repeater("e in EnterpriseCtrl.enterprises | orderBy:'name'").row(n))
+      .$$('a')
+      .click();
+  }
+
+  // navigate to the enterprise module before each test
+  beforeEach(function () {
+    browser.get(path);
+  });
+
+  it('Successfully creates a new Enterprise', function () {
+    FormUtils.buttons.create();
+
+    FormUtils.input('EnterpriseCtrl.enterprise.name', ENTERPRISES.name);
+    FormUtils.input('EnterpriseCtrl.enterprise.abbr', ENTERPRISES.abbr);
+    FormUtils.input('EnterpriseCtrl.enterprise.email', ENTERPRISES.email);
+    FormUtils.input('EnterpriseCtrl.enterprise.po_box', ENTERPRISES.po_box);
+    FormUtils.input('EnterpriseCtrl.enterprise.phone', ENTERPRISES.phone);
+
+    // select a random, non-disabled option
+    FormUtils.select('EnterpriseCtrl.enterprise.location_id')
+      .enabled()
+      .first()
+      .click();
+
+    FormUtils.radio('EnterpriseCtrl.enterprise.currency_id', ENTERPRISES.currency_id);  
+    // submit the page to the server
+    FormUtils.buttons.submit();
+  });
+
+
+  it('Successfully edits an Enterprise', function () {
+    element(by.id('enterprise-1')).click();
```

Comment:
Where did this ID come from?  Is this a hard-coded ID?  A better test should prove that you can edit _any_ enterprise, not just `id="enterprise-1"`.

To make this test better, ask the question: "How does the user select an enterprise to edit?"  (1) Do they click a button?  (2) Do they choose an enterprise from a list of enterprises?

In case (1), use protractor to find a (random) enterprise and click the button associated with it.  In case (2), use protractor to select a random enterprise from the list.